OS X 0.4-rc3: error: src/codegen.cpp: No such file or directory

```julia
julia> versioninfo()
Julia Version 0.4.0-rc3
Commit 483d548* (2015-09-27 20:34 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i5-2500S CPU @ 2.70GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3

julia> using SnoopCompile

julia> SnoopCompile.@snoop "/tmp/images_compiles.csv" begin
              1+2
              end
Turning on compiler logging
error: src/codegen.cpp: No such file or directory
make: *** No targets specified and no makefile found.  Stop.
ERROR: failed process: Process(`sh snoop.sh ''`, ProcessExited(2)) [2]
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,7 +4,6 @@ os:
   - linux
   - osx
 julia:
-  - release
   - nightly
 notifications:
   email: false

--- a/src/SnoopCompile.jl
+++ b/src/SnoopCompile.jl
@@ -27,8 +27,11 @@ then use `@snoop1`. When finished, restore the compiler to its
 original state with `snoop_off()`.
 """
 function snoop_on()
-    cd(snoop_path()) do
-        run(`sh snoop.sh ""`)
+    if VERSION < v"0.5.0-dev+210"
+        println("Turning on compiler logging")
+        cd(snoop_path()) do
+            run(`sh snoop.sh ""`)
+        end
     end
 end
 
@@ -38,11 +41,13 @@ state. Alternatively, you can just use `git checkout src/codegen.cpp
 && make` from the julia source folder.
 """
 function snoop_off()
-    cd(snoop_path()) do
-        run(`sh snoop.sh "--reverse"`)
+    if VERSION < v"0.5.0-dev+210"
+        println("Restoring compiler to original state")
+        cd(snoop_path()) do
+            run(`sh snoop.sh "--reverse"`)
+        end
     end
 end
-
 """
 ```
 @snoop "compiledata.csv" begin
@@ -60,7 +65,6 @@ manually with `snoop_on()` and `snoop_off()`, use `@snoop1` in a fresh
 julia session instead.
 """
 macro snoop(filename, commands)
-    println("Turning on compiler logging")
     snoop_on()
     try
         println("Launching new julia process to run commands...")
@@ -84,7 +88,6 @@ macro snoop(filename, commands)
         wait(pobj)
         println("done.")
     finally
-        println("Restoring compiler to original state")
         snoop_off()
     end
     nothing

--- a/test/REQUIRE
+++ b/test/REQUIRE
@@ -1,0 +1,1 @@
+ColorTypes

--- a/test/colortypes.jl
+++ b/test/colortypes.jl
@@ -1,0 +1,35 @@
+using SnoopCompile
+
+### Log the compiles
+# This only needs to be run once (to generate "/tmp/colortypes_compiles.csv")
+
+SnoopCompile.@snoop "/tmp/colortypes_compiles.csv" begin
+    include(Pkg.dir("ColorTypes", "test","runtests.jl"))
+end
+
+### Parse the compiles and generate precompilation scripts
+# This can be run repeatedly
+
+# IMPORTANT: we must have the module(s) defined for the parcelation
+# step, otherwise we will get no precompiles for the Colortypes module
+using ColorTypes
+
+data = SnoopCompile.read("/tmp/colortypes_compiles.csv")
+
+# Use these two lines if you want to create precompile functions for
+# individual packages
+pc, discards = SnoopCompile.parcel(data[end:-1:1,2])
+SnoopCompile.write("/tmp/precompile", pc)
+
+# Use these two lines if you want to add to your userimg.jl
+pc = SnoopCompile.format_userimg(data[end:-1:1,2])
+SnoopCompile.write("/tmp/userimg_ColorTypes.jl", pc)
+
+function notisempty(filename, minlength=1)
+    @test isfile(filename)
+    @test length(readlines(filename)) >= minlength
+end
+
+notisempty("/tmp/precompile/precompile_ColorTypes.jl", 100)
+notisempty("/tmp/precompile/precompile_FixedPointNumbers.jl", 2)
+notisempty("/tmp/userimg_ColorTypes.jl", 100)

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -1,5 +1,4 @@
 using SnoopCompile
 using Base.Test
 
-# write your own tests here
-@test 1 == 1
+include("colortypes.jl")

